helpers/routeHelpers.js

```diff
@@ -56,5 +58,22 @@ export const schema = {
     },
     id: Joi.object().keys({
         param: Joi.string().regex(/^[0-9a-fA-Z]{24}$/).required()
-    })
+    }),
+    user: {
+        post: Joi.object().keys({
+            login: Joi.string().required(),
+            password: Joi.string().required().regex(/^[a-zA-Z0-9]{8}$/),
+            role: Joi.array().required()

```

Comment:
Could you read about nested schema?
Best answer in these article
https://stackoverflow.com/questions/18001478/referencing-another-schema-in-mongoose